crates/rolldown/src/stages/link_stage/bind_imports_and_exports.rs

```diff
@@ -606,7 +602,21 @@ impl<'a> BindImportsAndExportsContext<'a> {
         }
         ImportStatus::_CommonJSWithoutExports => todo!(),
         ImportStatus::_Disabled => todo!(),
-        ImportStatus::External => todo!(),
+        ImportStatus::External(symbol_ref) => {
+          if self.options.format.keep_esm_import_export_syntax() {
+            // Imports from external modules should not be converted to CommonJS
+            // if the output format preserves the original ES6 import statements
+            break MatchImportKind::Normal { symbol: tracker.imported_as };
+          }
+
+          match &tracker.imported {
+            Specifier::Star => MatchImportKind::Namespace { namespace_ref: symbol_ref },
```

Comment:
Here esbuild using `namedImport.NamespaceRef`, ref https://github.com/evanw/esbuild/blob/main/internal/linker/linker.go#L2639. 

But it caused generate different external namespace, example 

``` .js
var import_external = require("external");
var import_external2 = require("external");
var import_external3 = __toESM(require("external"));
console.log(import_external.a, import_external2.b, import_external3.default);
```

So here i using `External#symbol_ref` to make it from one symbol.